build-logic/module-identity/src/main/kotlin/gradlebuild/identity/extension/ReleasedVersionsDetails.kt

```diff
@@ -41,10 +41,11 @@ class ReleasedVersionsDetails(currentBaseVersion: GradleVersion, releasedVersion
         }
 
         val latestFinalRelease = releasedVersions.finalReleases.first()
-        val latestRelease = listOf(releasedVersions.latestReleaseSnapshot, releasedVersions.latestRc).filter { it.gradleVersion() > latestFinalRelease.gradleVersion() }.maxByOrNull { it.buildTimeStamp() } ?: latestFinalRelease
+        val latestRelease = listOf(releasedVersions.latestReleaseSnapshot, releasedVersions.latestRc).filter { it.gradleVersion() > latestFinalRelease.gradleVersion() }.maxByOrNull { it.buildTimeStamp() }
+            ?: latestFinalRelease
         val previousVersions = (listOf(latestRelease) + releasedVersions.finalReleases).filter { it.gradleVersion() >= lowestInterestingVersion && it.gradleVersion().baseVersion < currentBaseVersion }.distinct()
         allPreviousVersions = previousVersions.map { it.gradleVersion() }
-        mostRecentRelease = previousVersions.first().gradleVersion()
+        mostRecentRelease = (listOf(releasedVersions.latestRc) + releasedVersions.finalReleases).first().gradleVersion()
```

Comment:
Why change this? It looks like the previous code made a lot of sense.

Reply:
In the previous code, `mostRecentRelease` and `mostRecentSnapshot` were same...

Reply:
Only if there is a release snapshot. If there isn't a release snapshot, then we'd use the the RC or whatever. Why is using a release snapshot bad?

Reply:
In which case there is no release snapshot? I think release snapshot (nightly) is always there so `mostRecentRelease` and `mostRecentSnapshot` is always same. Why do we keep two same fields?